### ISSUE TITLE
Allow warm_start

### ISSUE THREAD

**Issue by heidmic** [heidmic/suprb#66]
In some of our use cases, a limited set of rules (obtained through various means) already exists. Typically, these rules are insufficient to solve the problem by themselves, hence, the need for ML, but we assume, they are beneficial for solving it because they contain some information. They might also be slightly misplaced leading to poorer performance. However, solving this is much more complicated as one would need to incorporate the genetic information into rule discovery steps which is not straight forward.

tl;dr:
We want to create a set of rules in some other way than training, load them, start training based on them. In this issue only as part of the pool, later on also as potential parents for subsequent rule discovery steps (this is a much larger issue, yet to be created)

fit: pool is not wiped if warm_start is set (use sklearn notation) and can be initialised in some way, e.g. loading from json

**Comment by RomanSraj** [comment]
Possible with json load
